server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -182,7 +183,8 @@ public Translog(
         final LongSupplier globalCheckpointSupplier,
         final LongSupplier primaryTermSupplier,
         final LongConsumer persistedSequenceNumberConsumer,
-        final TranslogOperationHelper translogOperationHelper
+        final TranslogOperationHelper translogOperationHelper,
+        final ChannelFactory channelFactory
```

Comment:
@RajatGupta02 This is the reason that Detect Breaking Changes is failing. This PR adds a constructor arg to Translog. I think it can be fixed by doing the same pattern of LocalTranslog with 2 constructors.